### PR TITLE
Code changes for new OneAuth cache key migration API

### PR DESCRIPTION
**Pull request by iambmelt**
(Technically this is a major API break)

Reimplements the `ADAL#setSecretKey` migration API exposed for OneAuth.

Functions:
- Allows for behavioral control via `ReencryptionParams`
    * `abortOnFailure`
    * `eraseEntryOnFailure`
    * `eraseAllOnFailure`
- Data will not be written/overwritten unless explicitly overridden to do so
- Returns a new `IMigrationOperationResult` class to surface information about successes/failures

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/DefaultSharedPrefsFileManagerReencrypter.java [hunk 1]
```diff
@@ -1,96 +0,0 @@
...
-
-import com.microsoft.identity.common.internal.controllers.TaskCompletedCallbackWithError;
-
-import java.util.HashMap;
-import java.util.Map;
-
-/**
- * Default implementation of {@link ISharedPrefsFileManagerReencrypter}.
- */
-public class DefaultSharedPrefsFileManagerReencrypter implements ISharedPrefsFileManagerReencrypter {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/DefaultSharedPrefsFileManagerReencrypter.java:34]: Why did we move this class?

File: common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        }
+    }
+
+    @Override
+    public void reencryptAsync(@NonNull final ISharedPreferencesFileManager fileManager,
+                               @NonNull final IStringEncrypter encrypter,
+                               @NonNull final IStringDecrypter decrypter,
+                               @NonNull final ReencryptionParams params,
+                               @NonNull final TaskCompletedCallback<IMigrationOperationResult> callback) {
+        new Thread(new Runnable() {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java:160]: I must have asked this before, but can we put this in a pool?

File: common/src/main/java/com/microsoft/identity/common/migration/MigrationOperationResult.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+/**
+ * The result of a cache reencryption operation.
+ */
+@Accessors(prefix = "m")
+class MigrationOperationResult implements IMigrationOperationResult {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/migration/MigrationOperationResult.java:39]: I'd personally have this class be immutable.

File: common/src/main/java/com/microsoft/identity/common/internal/cache/DefaultSharedPrefsFileManagerReencrypter.java (hunk 1 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/DefaultSharedPrefsFileManagerReencrypter.java:34]: Will make these changes additive in next iteration

File: common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java (hunk 2 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java:160]: Can do

**Pull request closed without merging**